iceoryx_utils/test/integrationtests/test_resizeable_lockfree_queue_stresstest.cpp

```diff
@@ -0,0 +1,829 @@
+// Copyright (c) 2020 Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "test.hpp"
+
+#include "iceoryx_utils/concurrent/resizeable_lockfree_queue.hpp"
+
+// Remark: It would be nice to have way to configure the (maximum) runtime in a general way.
+
+using namespace ::testing;
+
+#include <atomic>
+#include <list>
+#include <numeric>
+#include <random>
+#include <thread>
+#include <utility>
+#include <vector>
+
+namespace
+{
+struct Data
+{
+    Data(int id = 0, size_t count = 0)
```

Comment:
I disagree here. It is perfectly valid for the client to use plain ints in his Data structure stored in the queue, it should work regardless.

I can change it though, but the queue has no such restricting on the types it works with.

Reply:
There might be issues because various platforms might have different width for this. `char` for example can be 10 bit wide on some platforms. It might save your future self from fixing a tests.

Reply:
Since it will work with uint64_t as well I can change it. Nevertheless we need still use plain ints in many tests and should think about valid data types for tests and state this clearly. I myself think it should be less restrictive as the actual library code, we e.g. use dynamic memory in tests as well.

Reply:
fully agree with the dynamic memory in tests

Reply:
uint64_t now